src/platform/Zephyr/BLEManagerImpl.cpp

```diff
@@ -790,10 +789,16 @@ void BLEManagerImpl::HandleTXIndicated(struct bt_conn * conId, bt_gatt_indicate_
 void BLEManagerImpl::HandleConnect(struct bt_conn * conId, uint8_t err)
 {
     ChipDeviceEvent event;
+    struct bt_conn_info bt_info;
 
     PlatformMgr().LockChipStack();
 
+    sBleConnNum++;
+    ChipLogProgress(DeviceLayer, "Current number of connections: %u/%u", sBleConnNum, CONFIG_BT_MAX_CONN);
+
     // Don't handle BLE connecting events when it is not related to CHIPoBLE
+    VerifyOrExit(bt_conn_get_info(conId, &bt_info) == 0, );
+    VerifyOrExit(bt_info.role == BT_CONN_ROLE_PERIPHERAL, );
```

Comment:
I'd suggest to add a comment why we drop non-peripheral connections.

Reply:
Ok, I thought it's clear based on the comment above (Don't handle BLE connecting events when it is not related to CHIPoBLE) but I can further clarify that according to Matter spec we have to be a peripheral.